21-Hello-World/01-FP-Philosophical-Foundations/05-Type-Classes.md

```diff
@@ -21,23 +21,92 @@ functionName Apple = "Apple"
+    - **Note: Some laws cannot always be enforced.**
 3. (Frequently) The functions/values that can be derived once one implements the type class.
     - Most of the power/flexibility of type classes come from the combination of the main functions/values implemented in a type class' definition and these derived functions. When a type class extends another, the type class' power increases, its flexibility decreases, and its costs increase.
         - For example, `Apply` is a less powerful typeclass than `Monad` because it does not have `bind` but is more flexible than `Monad` because it can compute things in parallel.
-4. (Sometimes) Something that combines multiple type classes together into one.
 
-While it is possible for a programmer to define a type class without laws, there is a debate about whether this is a good decision. Most say classes should always have laws (i.e. be a term from Category Theory) whereas others say laws should not always be required because it enables reusing the same function name for the same concept (i.e. be 'just' an interface). The counterargument to this last point is that one usually hasn't thought through their design that deeply yet. A Reddit thread discusses some of the tradeoffs of each approach here: https://www.reddit.com/r/haskell/comments/5gospp/dont_use_typeclasses_to_define_default_values/
+Some type classes (e.g. `ParentTypeClass`) combine two or more other type classes (e.g. `ChildTypeClass`). **This parent-child-like relationship is not necessarily hierarchial (type must satisfy child before parent possibility exists) but synchronous (type must satisfy both child and parent)**. This leads to the following possibilities:
+- `ParentTypeClass` asserts that some type has an instance for of all its `ChildTypeClass`es.
+    - Some derived functions for `ParentTypeClass` are only possible if the implementation can use functions from two or more `ChildTypeClass`es
+    - Some implementations of `ChildTypeClass` can be done more easily / better by using functions/values from `ParentTypeClass`
+- `ParentTypeClass` forces implementations for `ChildTypeClass` to satisfy additional law(s).
+
+In other words, if a type has an instance for some class X and another class Y
+
+## Dictionaries and Lawless Type Classes
+
+Dictionaries are what enable a function/value to magically appear in the implementation of a function's body. Read [this article about type class 'dictionaries'](https://www.schoolofhaskell.com/user/jfischoff/instances-and-dictionaries) as there really is no better way to explain this concept.
+
+After reading the above article, it's clear that this code...
+```purescript
+class ToBoolean a where
+  toBoolean :: a -> Boolean
+
+  unUsed :: a -> String
+
+example :: forall a. ToBoolean a => a -> Boolean
+example value = toBoolean value
+```
+
+... gets desugared to this code
+
+```purescript
+type ToBooleanDictionary a =
```

Comment:
shouldn't this be `data ToBooleanDictionary a`?

Reply:
Ah, you're right. Whoops!